lib/msprime.c

```diff
@@ -807,53 +807,147 @@ msp_verify_segments(msp_t *self, bool verify_breakpoints)
     }
 }
 
-/* FIXME logic doesn't make sense in continuous space */
+typedef struct {
+    double seq_length;
+    segment_t *overlaps;
+} overlap_counter_t;
+
+static overlap_counter_t*
+overlap_counter_alloc(double seq_length, int initial_count)
+{
+    overlap_counter_t *counter = malloc(sizeof(overlap_counter_t));
```

Comment:
Just for the sake of keeping things consistent, it's probably better to make the first parameter of this ``self``, and the return type an int. Future readers might wonder why this object is following a different convention to other objects, when it doesn't really matter.